fendermint/vm/interpreter/src/genesis.rs

```diff
@@ -0,0 +1,795 @@
+        // This effectively burns the miner rewards. Better than panicking.
+        state
+            .create_builtin_actor(
+                account::ACCOUNT_ACTOR_CODE_ID,
+                reward::REWARD_ACTOR_ID,
+                &account::State {
+                    address: reward::REWARD_ACTOR_ADDR,
+                },
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create reward actor")?;
+
+        // STAGE 1b: Then we initialize the in-repo custom actors.
+
+        // Initialize the chain metadata actor which handles saving metadata about the chain
+        // (e.g. block hashes) which we can query.
+        let chainmetadata_state = fendermint_actor_chainmetadata::State::new(
+            &state.store(),
+            fendermint_actor_chainmetadata::DEFAULT_LOOKBACK_LEN,
+        )?;
+        state
+            .create_custom_actor(
+                fendermint_actor_chainmetadata::CHAINMETADATA_ACTOR_NAME,
+                chainmetadata::CHAINMETADATA_ACTOR_ID,
+                &chainmetadata_state,
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create chainmetadata actor")?;
+
+        let eam_state = fendermint_actor_eam::State::new(
+            state.store(),
+            PermissionModeParams::from(genesis.eam_permission_mode),
+        )?;
+        state
+            .replace_builtin_actor(
+                eam::EAM_ACTOR_NAME,
+                eam::EAM_ACTOR_ID,
+                fendermint_actor_eam::IPC_EAM_ACTOR_NAME,
+                &eam_state,
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to replace built in eam actor")?;
+
+        // STAGE 2: Create non-builtin accounts which do not have a fixed ID.
+
+        // The next ID is going to be _after_ the accounts, which have already been assigned an ID by the `Init` actor.
+        // The reason we aren't using the `init_state.next_id` is because that already accounted for the multisig accounts.
+        let mut next_id = init::FIRST_NON_SINGLETON_ADDR + addr_to_id.len() as u64;
+
+        for a in genesis.accounts {
+            let balance = a.balance;
+            match a.meta {
+                ActorMeta::Account(acct) => {
+                    state
+                        .create_account_actor(acct, balance, &addr_to_id)
+                        .context("failed to create account actor")?;
+                }
+                ActorMeta::Multisig(ms) => {
+                    state
+                        .create_multisig_actor(ms, balance, &addr_to_id, next_id)
+                        .context("failed to create multisig actor")?;
+                    next_id += 1;
+                }
+            }
+        }
+
+        // STAGE 3: Initialize the FVM and create built-in FEVM actors.
+
+        state
+            .init_exec_state(
```

Comment:
I know this is not something you have implemented in this PR but the fact the state is using the stage enum is really confusing.